### DISCUSSION TITLE
why the same chunk has difference hash in dynamic load

### DISCUSSION THREAD

**Discussion by WxSwen** [webpack/webpack#13337] [General]
# Bug report

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**
A and B module build as a chunk，and splitChunks set chunk is 'all'. But import by two entry, one is used async way, but another is used sync way. It will product one chunk with a hash, but async way has another hash without chunk.

just like this:
`optimization: {
    namedModules: true,
    namedChunks: true,
    moduleIds: 'hashed',
    chunkIds: 'named',
    splitChunks: {
      cacheGroups: {
'react-data': {
          chunks: 'all',
          test: module => {
            return /[\\/](redux|react-redux|react-router-redux|redux-immutable|redux-thunk|immutable|immutablediff|immutablepatch)([\\/]|$)/.test(module.context);
          },
          name: 'vendor-react-data',
          enforce: true,
          priority: 10,
          reuseExistingChunk: true
        }
}
}
}`

webpack version:  "webpack": "4.16.0",

**What is the expected behavior?**

I think, it should be product one chunk and hash be consistent

**Other relevant information:**
webpack version:  "4.16.0"
Node.js version: 
Operating System: 
Additional tools:

**Comment by alexander-akait** [comment]
Please use `Dicussion` for questions. I think it is bug in webpack, Can you update webpack version? 4.16 is very old
